servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1478,6 +1518,179 @@ public void setState(long fileId, SetStateOptions options)
+      return IdUtils.INVALID_WORKER_ID;
+    } catch (InvalidPathException e) {
+      LOG.error("The file {} to persist does not exist", fileId);
+      return IdUtils.INVALID_WORKER_ID;
+    }
+
+    if (workers.size() == 0) {
+      LOG.error("The file " + fileId + " does not exist on any worker");
+      return IdUtils.INVALID_WORKER_ID;
+    } else if (workers.size() > 1) {
+      LOG.info("The file is stored at more than one worker: " + workers);
+    }
+
+    // return the first worker that has all the blocks
+    for (Entry<Long, Integer> entry : workers.entrySet()) {
+      if (entry.getValue() == totalBlockNum) {
+        return entry.getKey();
+      }
+    }
+
+    LOG.error("Not all the blocks of file {} stored on the same worker", fileId);
+    return IdUtils.INVALID_WORKER_ID;
+  }
+
+  /**
+   * Polls the files to send to the given worker for persistence.
+   *
+   * @param workerId the worker id
+   * @return the list of files
+   * @throws FileDoesNotExistException if the file does not exist
+   * @throws InvalidPathException if the path is invalid
+   */
+  private List<PersistFile> pollToCheckpoint(long workerId)
+      throws FileDoesNotExistException, InvalidPathException {
+    List<PersistFile> files = Lists.newArrayList();
+    synchronized (mWorkerToAsyncPersistFile) {
+      if (!mWorkerToAsyncPersistFile.containsKey(workerId)) {
+        return files;
+      }
+    }
+
+    List<Long> toRequestFilePersistence = Lists.newArrayList();
+    synchronized (mWorkerToAsyncPersistFile) {
+      Set<Long> scheduledFiles = mWorkerToAsyncPersistFile.get(workerId);
+      for (long fileId : Sets.newHashSet(scheduledFiles)) {
+        InodeFile inode = (InodeFile) mInodeTree.getInodeById(fileId);
+        if (inode.isCompleted()) {
+          toRequestFilePersistence.add(fileId);
+          List<Long> blockIds = Lists.newArrayList();
+          for (FileBlockInfo fileBlockInfo : getFileBlockInfoList(fileId)) {
+            blockIds.add(fileBlockInfo.blockInfo.blockId);
+          }
+
+          PersistFile toCheckpoint = new PersistFile(fileId, blockIds);
+          files.add(toCheckpoint);
+          // update the inode file persisence state
+          inode.setPersistenceState(PersistenceState.IN_PROGRESS);
+          scheduledFiles.remove(fileId);
+        }
+      }
+    }
+
+    setPersistingState(toRequestFilePersistence);
+    // write to journal
+    PersistFilesRequestEntry persistFilesRequest =
+        PersistFilesRequestEntry.newBuilder().addAllFileIds(toRequestFilePersistence).build();
+    writeJournalEntry(
+        JournalEntry.newBuilder().setPersistFilesRequest(persistFilesRequest).build());
+    flushJournal();
+    return files;
+  }
+
+  /**
+   * Updates a list of files as being persisted.
+   *
+   * @param fileIds the id of the files
+   * @throws FileDoesNotExistException when a file does not exist
+   */
+  private void setPersistingState(List<Long> fileIds) throws FileDoesNotExistException {
```

Comment:
It's named so because the method only sets the persistent state to `IN_PROGRESS`